File: src/backend/src/main/java/com/bouh/backend/model/Dto/DoctorSearchDTO.java
package com.bouh.backend.model.Dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Data

public class DoctorSearchDTO {
    private String id;
    private String name;
    private String areaOfKnowledge;
    private double averageRating;
    private String profilePhotoURL;
}

File: src/frontend/lib/dto/DoctorSearchDto.dart
class DoctorSearchDTO {
  final String id;
  final String name;
  final String areaOfKnowledge;
  final double averageRating;
  final String profilePhotoURL;

  DoctorSearchDTO({
    required this.id,
    required this.name,
    required this.areaOfKnowledge,
    required this.averageRating,
    required this.profilePhotoURL,
  });

  factory DoctorSearchDTO.fromJson(Map<String, dynamic> json) {
    return DoctorSearchDTO(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      areaOfKnowledge: json['areaOfKnowledge'] ?? '',
      averageRating: (json['averageRating'] ?? 0).toDouble(),
      profilePhotoURL: json['profilePhotoURL'] ?? '',
    );
  }
}
